src/ScriptBuilder/Meters/MeterDefinitionReader.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace NServiceBus.Metrics.PerformanceCounters
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Mono.Cecil;
+
+    static class MeterDefinitionReader
+    {
+        public static bool TryGetMeterDefinition(TypeDefinition type, out List<MeterDefinition> definitions)
+        {
+            definitions = new List<MeterDefinition>();
+            if (type.BaseType != null && type.BaseType.FullName == "NServiceBus.Metrics.MetricBuilder")
+            {
+                var meterAttributes = type.Fields.Select(f => f.GetSingleAttribute("NServiceBus.Metrics.MeterAttribute")).Where(c => c != null);
+
+                foreach (var meterAttribute in meterAttributes)
+                {
+                    var name = (string)meterAttribute.ConstructorArguments[0].Value;
```

Comment:
@WojcikMike you on.y have access to what is expressed in the code. not the evaluated runtime. so cecil is essentially "looking at the code used to construct the attributive". which in this case is setting constructor args. if you want to look at props at il time you would need to change the attribute to remove the constructor and have those props as public sets.

BTW you should be able to access the constructor args by name if this is what you mean by "do you know of any better way"